spark/dl/src/main/scala/com/intel/analytics/bigdl/models/resnet/Train.scala

```diff
@@ -67,17 +67,12 @@ object Train {
         curModel
       }
 
-      val state = if (param.stateSnapshot.isDefined) {
-        T.load(param.stateSnapshot.get)
+      val optimMethod = if (param.stateSnapshot.isDefined) {
+        OptimMethod.load[Float](param.stateSnapshot.get)
```

Comment:
Now, the snapshot perserves the whole OptimMethod object, the object contains: necessary hyper parameters(learningRate...) and internal parameters(epoch, neval).